SimpleRss/SimpleRss/ViewControllers/HomeViewController.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  ViewController.swift
+//  SimpleRss
+//
+//  Created by Voldem on 11/14/18.
+//  Copyright © 2018 Vladimir Koptev. All rights reserved.
+//
+
+import UIKit
+
+class HomeViewController: UIViewController
+{
+    var tableView: UITableView?
+    var dataSource: TopicsSource?
+    var tableDelegate: TopicsDelegate?
+    
+    let topicCellIdentifier = "TopicViewCell"
+    
+    let topics: [Topic] = [Topic(id: 0, title: "IT", url: "https://img.tyt.by/n/brushko/0e/9/perseidy_12082017_tutby_brush_phsl_-9131.jpg"),
+                                 Topic(id: 1, title: "Economics", url: "https://img.tyt.by/n/01/a/mid_belarusi_st.jpg"),
+                                 Topic(id: 2, title: "Politics", url: "https://img.tyt.by/n/it/0f/7/world-of-tanks.jpg")]
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView = UITableView(frame: self.view.bounds)
+        dataSource = TopicsSource(context: self)
+        tableDelegate = TopicsDelegate(context: self)
+        tableView?.dataSource = dataSource
+        tableView?.delegate = tableDelegate
+        
+        let nib = UINib(nibName: topicCellIdentifier, bundle: nil)
+        
+        tableView?.register(nib, forCellReuseIdentifier: topicCellIdentifier)
+        tableView?.estimatedRowHeight = 50
+        tableView?.rowHeight = UITableView.automaticDimension
+        tableView?.separatorStyle = UITableViewCell.SeparatorStyle.none
+
+        self.view.addSubview(tableView!)
+        tableView?.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.title = "rss"
+    }
+}
+
+internal class TopicsDelegate: NSObject, UITableViewDelegate
+{
+    weak var context: HomeViewController?
+    
+    init(context: HomeViewController?) {
+        self.context = context
+    }
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath){
+        context?.navigationController?.pushViewController(FeedViewController(name: indexPath.row), animated: true)
```

Comment:
I think here Topics delegate class depends on context **having** navigation controller to work properly, and I also think this is the case where should take [DIP](https://en.wikipedia.org/wiki/Dependency_inversion_principle) and [Open-Closed principle into consideration](https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle)